### PR TITLE
Refactor: update prettier configuration and refactor accordingly

### PR DESCRIPTION
**Pull request by s-pace**
This PR:
- uses same prettier configuration as autocomplete & InstantSearch
- reformat every file accordingly

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-      {
-        files: '*.md',
-        options: {
-          parser: 'markdown',
-          printWidth: 80,
-          proseWrap: 'always',
-          singleQuote: true,
-          trailingComma: 'es5',
-        },
+  overrides: [
```
Comment (francoischalifour) [inline .prettierrc.js]: Do we need this override? The only different value is `proseWrap`, which could be relied upon by the editors settings ("View > Toggle word wrap" in VSCode).

Comment (s-pace) [reply .prettierrc.js]: IMHO the configuration should work with the IDE and the prettier CLI so yes I think we should still have it from the configuration.
As for overriding or not, I actually do not know, I only kept the configuration which was already in place. I will have a look.

Comment (francoischalifour) [reply .prettierrc.js]: If we want to keep `proseWrap`, we can get rid of the `overrides` key and set the top-level `proseWrap` value to `"always"` then.

Comment (s-pace) [reply .prettierrc.js]: Make sense. I have updated the codebase and the configuration accordingly. Please review again

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**